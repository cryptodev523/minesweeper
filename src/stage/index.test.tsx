import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { cleanup } from '@testing-library/react';

import Board from './index';

describe('Stage', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('should render correctly and match with snapshot', () => {
    const map: string[] = [];
    const tree = renderer.create(<Board map={map} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to find at least 1 square cell', () => {
    const map: string[] = [
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
      '□□□□□□□□□□',
    ];
    const wrapper = mount(<Board map={map} />);
    const element = wrapper.find({ 'data-testid': 'cell-1-1' });

    expect(element).toBeTruthy();
  });
});
