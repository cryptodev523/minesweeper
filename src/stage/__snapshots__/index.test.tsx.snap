// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage should render correctly and match with snapshot 1`] = `
<div
  className="makeStyles-board-1 MuiBox-root css-0"
/>
`;
